src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts

```diff
@@ -57,12 +60,18 @@ export class TreeSelectionHandler {
 		let specificSelection = selection[0];
 
 		if (this.isMouseEvent(event)) {
-			if (this._lastClicked === specificSelection) {
+			if (this._lastClicked !== undefined) {
 				clearTimeout(this._clickTimer);
-				sendSelectionEvent(event, selection, true);
-				return;
+				let lastSpecificClick = this._lastClicked[0];
+
+				if (lastSpecificClick === specificSelection) {
+					sendSelectionEvent(event, selection, true);
+					return;
+				} else {
+					sendSelectionEvent(event, this._lastClicked, false);
+				}
 			}
-			this._lastClicked = specificSelection;
+			this._lastClicked = selection;
 		}
 
 		this._clickTimer = setTimeout(() => {
```

Comment:
Does it make more sense to just move the setTiemout to the if mouse event since for other selection types (keyboard) we don't want to use this timeout method?

Reply:
Hmm yeah. Do we want to treat a keyboard selection as a full double click?